File: cp_rust/src/bin/cses-1617-bit-strings.rs
// Created by Ayush Biswas at 2025/07/10 12:01
// https://cses.fi/problemset/task/1617
use cp_lib::*;

// @code begin
use cpio::*;
use math::pow_mod;

sol! {
    fn solution(
        n: i64,
    ) -> u32 {
        pow_mod(2, n, 1e9 as u32 +7)
    }
}

// @code end

File: cp_haskell/bin/cses-1617-bit-strings.hs
-- Created by Ayush Biswas at 2026/03/04 10:07
-- https://cses.fi/problemset/task/1617
-- @code begin

m :: Integer
m = 10 ^ 9 + 7

main :: IO ()
main = do
  n :: Int <- read <$> getLine
  let res = (2 ^ n) `mod` m
  print res

-- @code end